gpsrenda/bin/renda

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+import argparse as ap
+from datetime import datetime
+from datetime import timedelta
+from glob import glob
+from importlib import import_module
+from os import makedirs
+from os.path import basename, dirname, join
+
+import cairo
+import moviepy.editor as mpy
+import numpy as np
+import yaml
+
+from gpsrenda.datasources import FitDataSource
+from gpsrenda.utils import extract_start_time, timestamp_to_seconds
+
+def renda(video_paths, data_path, time_offset, config_path, preview):
+    data_source = FitDataSource(data_path)
+
+    with open(config_path, 'r') as config_file:
+        config_data = yaml.safe_load(config_file)
+
+    widgets = []
+    widget_module = import_module('gpsrenda.widgets.widgets')
```

Comment:
another way to do this might be to have a `gpsrenda.widgets.allwidgets` array, and have each widget call like a `gpsrenda.widgets.register_widget()` function.  that way you could have other packages with widgets that you could simply `import` to use (or have the package name of in a conf file).  and, then, for bonus points, the `type:` could then be lowercase... because cool kids use lowercase.

Reply:
I know there are patterns for user-supplied modules but I think of this more like a programmatic interface to something like Virb Edit which has a total sufficient set of widgets on its own. I think it's easier for us to go with "if you want to see a new widget submit a PR".